src/transformers/testing_utils.py

```diff
@@ -614,7 +614,11 @@ def require_torch_multi_npu(test_case):
     # Set env var CUDA_VISIBLE_DEVICES="" to force cpu-mode
     import torch
 
-    if torch.cuda.is_available():
+    if "TRANSFORMERS_TEST_DEVICE" in os.environ:
+        torch_device = os.environ["TRANSFORMERS_TEST_DEVICE"]
+        # try creating device to see if provided device is valid
+        _ = torch.device(torch_device)
```

Comment:
Maybe a nice try except here to throw a helpful error message? But yes better this way!